src/liballoc/boxed.rs

```diff
@@ -125,6 +153,111 @@ impl<T> Box<T> {
     }
 }
 
+impl<T> Box<[T]> {
+    /// Construct a new boxed slice with uninitialized contents.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(new_uninit)]
+    ///
+    /// let mut values = Box::<[u32]>::new_uninit_slice(3);
+    ///
+    /// let values = unsafe {
+    ///     // Deferred initialization:
+    ///     values[0].as_mut_ptr().write(1);
+    ///     values[1].as_mut_ptr().write(2);
+    ///     values[2].as_mut_ptr().write(3);
+    ///
+    ///     values.assume_init()
+    /// };
+    ///
+    /// assert_eq!(*values, [1, 2, 3])
+    /// ```
+    #[unstable(feature = "new_uninit", issue = "0")]
+    pub fn new_uninit_slice(len: usize) -> Box<[mem::MaybeUninit<T>]> {
+        let layout = alloc::Layout::array::<mem::MaybeUninit<T>>(len).unwrap();
+        let ptr = unsafe { alloc::alloc(layout) };
+        let unique = Unique::new(ptr).unwrap_or_else(|| alloc::handle_alloc_error(layout));
+        let slice = unsafe { slice::from_raw_parts_mut(unique.cast().as_ptr(), len) };
+        Box(Unique::from(slice))
+    }
+}
+
+impl<T> Box<mem::MaybeUninit<T>> {
```

Comment:
You could move the `new_uninit` into this `impl` block I think, that seems like a nioce way to group these methods together.

Reply:
This would make a call like `Box::<u32>::new_uninit()` becomes `Box::<MaybeUnint<u32>>::new_uninit()` which feels unnecessarily verbose (though technically more accurate, since it matches the return type).

I suspect that this turbofish will be needed in a lot of cases to appease type inference.

Reply:
Oh I see. Fine for me either way, but can we make sure that this gets talked about when stabilization is discussed? Like, mention it in the tracking issue or so?

Reply:
Opened https://github.com/rust-lang/rust/issues/63291 with a note of this.

Reply:
Opened #63291 with a note of this.